pkg/virtual/apiexport/authorizer/maximal_permission_policy.go

```diff
@@ -0,0 +1,176 @@
+	newDeepSARAuthorizer    func(clusterName logicalcluster.Name) (authorizer.Authorizer, error)
+	getAPIExportsByIdentity func(identityHash string) ([]*apisv1alpha1.APIExport, error)
+}
+
+// NewMaximalPermissionAuthorizer creates an authorizer that checks the maximal permission policy
+// for the requested resource if the resource is a claimed resource in the requested API export.
+// The check is omitted if the requested resource itself is not associated with another API export.
+//
+// If the request is a cluster request the authorizer skips authorization if the request is not for a bound resource.
+// If the request is a wildcard request this check is skipped because no unique API binding can be determined.
+func NewMaximalPermissionAuthorizer(deepSARClient kubernetes.ClusterInterface, apiExportInformer apisinformers.APIExportInformer, apiBindingInformer apisinformers.APIBindingInformer) authorizer.Authorizer {
+	apiExportLister := apiExportInformer.Lister()
+	apiExportIndexer := apiExportInformer.Informer().GetIndexer()
+
+	return &maximalPermissionAuthorizer{
+		getAPIExport: func(clusterName, apiExportName string) (*apisv1alpha1.APIExport, error) {
+			return apiExportLister.Get(clusters.ToClusterAwareKey(logicalcluster.New(clusterName), apiExportName))
+		},
+		getAPIExportsByIdentity: func(identityHash string) ([]*apisv1alpha1.APIExport, error) {
+			return indexers.ByIndex[*apisv1alpha1.APIExport](apiExportIndexer, indexers.APIExportByIdentity, identityHash)
+		},
+		newDeepSARAuthorizer: func(clusterName logicalcluster.Name) (authorizer.Authorizer, error) {
+			return delegated.NewDelegatedAuthorizer(clusterName, deepSARClient)
+		},
+	}
+}
+
+func (a *maximalPermissionAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+	apiDomainKey := dynamiccontext.APIDomainKeyFrom(ctx)
+	parts := strings.Split(string(apiDomainKey), "/")
+	if len(parts) < 2 {
+		return authorizer.DecisionNoOpinion, VirtualAPIExportNotPermittedReason, fmt.Errorf("invalid API domain key")
+	}
+
+	claimingAPIExportCluster := parts[0]
+	claimingAPIExportName := parts[1]
+
+	claimingAPIExport, err := a.getAPIExport(claimingAPIExportCluster, claimingAPIExportName)
+	if kerrors.IsNotFound(err) {
+		return authorizer.DecisionNoOpinion, VirtualAPIExportNotPermittedReason, fmt.Errorf("API export not found: %w", err)
+	}
+
+	claimedIdentityHash, found := getClaimedIdentity(claimingAPIExport, attr)
+	if !found {
+		// it's a resource in the claiming API export, hence unclaimed
+		return authorizer.DecisionAllow, "unclaimed resource", nil
+	}
+	if claimedIdentityHash == "" {
+		// it's a native k8s resource (secret, configmap, ...), or a system kcp CRD resource (apis.kcp.dev)
+		// For neither case a maximum permission policy can exist.
+		return authorizer.DecisionAllow, "unclaimable resource", nil
+	}
+
+	claimedAPIExports, err := a.getAPIExportsByIdentity(claimedIdentityHash)
```

Comment:
nit: this code is super dense so I think for clarity variable names will make a big difference - I got lost with `claimedAPIExports` - since there's no concept of claiming an export. This is really the list of exports that provide the APIs which then have permission claims on them ... no idea what to call this. `apiExportsProvidingClaimedResources`?

Reply:
`apiExportsProvidingClaimedResources` sounds very verbose but if it helps i can rename this one 👍

Reply:
I don't have strong feelings on the particular name :) --> maybe the Germans will like that long one more though :D